src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,109 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
+
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
 
-export const NewMovie = () => {
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```

Comment:
you can merge this useStates into one
```suggestion
  const [newMovie, setNewMovie] = useState({
     title: '',
     description: '',
     ...
  })
```

Reply:
For changing proper field in this useState you can use handle function
```
  const handleChange = (name: keyof Movie, value: string) => {
    setNewMovie(prev => ({
      ...prev,
      [name]: value,
    }));
  };
```

Reply:
I have done changes, but in my opinion its became more complicate to understand) especially with function handleChange